hdr_html_canvas_element.md

```diff
@@ -1,749 +1,336 @@
-  const g3 = simpleInverseTransform(g2, systemGamma); 
-  const b3 = simpleInverseTransform(b2, systemGamma); 
-  const [r4, g4, b4] = limitTosRGBGamut(r3, g3, b3);
-  return [r4, g4, b4];
-}
+```idl
+  enum CanvasColorType {
+    "unorm8",
+    "float16",
+  };
 ```
 
-##### extended sRGB signal
+When `colorType = "unorm8"`, the non-linear component signals {R', G', B'} are
+quantized using full range quantization, i.e. they are multiplied by 255 and
+rounded to the nearest integer. For example, R' = 0.5 is represented by the
+integer value 128.
 
-Extended sRGB contains information that is outside of the capabilities of an sRGB monitor.  For this reason, the extended sRGB must first be converted for display.  
-The extended sRGB signal can be converted for display on an sRGB monitor by first converting to HLG or PQ and then applying the relevant tonemapping to sRGB for displays.  For example:
-```javascript
-const [r_display, g_display, b_display] = tonemapREC2100HLGtoSRGBdisplay(
-       convertExtendedSRGBtoREC2100HLG(r_extended_srgb, g_extended_srgb, b_extended_srgb));
-```
+When `colorType = "float16"`, the non-linear component signals {R', G', B'} are
+not quantized, i.e. R' = 0.5 is represented by the floating point value 0.5.
 
-### Color spaces
+_NOTE: `colorType = "unorm8"` corresponds to HTML Canvas as it exists today and
+should not be used to represent HDR signal, as detailed in the introduction._
 
-#### General
+## Extend `ImageDataSettings` to support higher bit depths
 
-Update `PredefinedColorSpace` to include the following color spaces.
+Add to `ImageDataSettings` a `ImageDataColorType` member that specifies the
+conversion semantics and type of each of the items of the `data` member array of
+`ImageData`.
 
 ```idl
-  partial enum PredefinedColorSpace {
-    'srgb-linear',
-    'rec2100-hlg',
-    'rec2100-pq',
-  }
+  partial dictionary ImageDataSettings {
+    ImageDataColorType colorType = "Uint8ClampedArray";
+  };
 ```
 
-#### srgb and display-p3
-
-There already exist defined color spaces `srgb` and `display-p3`.
+```idl
+  enum ImageDataColorType {
+    "unorm8",
+    "float16",
+    "float32"
+    // and potentially others
+  };
+```
 
-Note that the transfer function for these spaces [is already defined](https://www.w3.org/TR/css-color-4/#predefined) on all real numbers (not just the unit interval), as:
+The values `"unorm8"`, `"float16"` and `"float32"` result in `data` returning an
+array with the type `Uint8ClampedArray`, `Float16Array`, `Float32Array`,
+respectively.
 
-```javascript
-  function electroOpticalTransferFunction(c) {
-    let sign = c < 0? -1 : 1;
-    let abs = Math.abs(c);
-    if (abs <= 0.04045) {
-      return = c / 12.92;
-    }
-    else {
-      return = sign * (Math.pow((abs + 0.055) / 1.055, 2.4));
-    }
-  }
-```
+## Add HDR rendering behavior and HDR metadata to `CanvasRenderingContext2DSettings`
 
-#### srgb-linear
+Add a new CanvasColorMetadata dictionary:
```

Comment:
How will we help developers provide good values for the dataype and colorspace of the canvas?  I originally thought the spec would have recommendations based on the metadata but the metadata is only available after the canvas is created.  

Do we expect developers to first make an sacrificial canvas, see what metadata is returned, and then make the "real" canvas afterwards?  Seems like the metadata should be available independent of an individual canvas instance. 

In general, I think the spec should provide use cases for the metadata such as how the metadata guides what `DOMString` values to pass for 2D canvas colors. Without use cases, this is going to draw the ire of those who oppose any fingerprinting bits be exposed to web developers for informational purposes. I, personally, think we can make the case for having these but we need to be more clear why it is needed in the spec.

Reply:
Yes, we definitely need more information -- similar to what is provided in the PNG 3ED spec: https://w3c.github.io/PNG-spec/#mDCv-chunk and https://w3c.github.io/PNG-spec/#cLLi-chunk

There are two basic use cases:

* the application receives the ST 2086 metadata, e.g. from metadata within the image file or from an API call. In this case, the application simply copies the information as in the `CanvasSmpteSt2086Metadata` structure
* the application generates the ST 2086 metadata, in which case, there are best practices we should be able to point to

Reply:
See added prose